src/main/java/uk/gov/hmcts/reform/wacaseeventhandler/services/EventMessageReceiverService.java

```diff
@@ -0,0 +1,134 @@
+            log.error("Could not parse the message with id '{}'", messageId);
+
+            CaseEventMessageEntity messageEntity = build(messageId, message, fromDlq, MessageState.UNPROCESSABLE);
+            repository.save(messageEntity);
+
+            return mapper.mapToCaseEventMessage(messageEntity);
+        } catch (DataIntegrityViolationException e) {
+            throw new CaseEventMessageDuplicateMessageIdException(
+                format("Trying to save a message with a duplicate messageId: %s", messageId), e);
+        }
+    }
+
+    private CaseEventMessageEntity build(String messageId,
+                                         String message,
+                                         boolean fromDlq,
+                                         EventInformation eventInformation,
+                                         MessageState state) throws JsonProcessingException {
+        CaseEventMessageEntity caseEventMessageEntity = new CaseEventMessageEntity();
+        caseEventMessageEntity.setMessageId(messageId);
+        caseEventMessageEntity.setCaseId(eventInformation.getCaseId());
+        caseEventMessageEntity.setEventTimestamp(eventInformation.getEventTimeStamp());
+        caseEventMessageEntity.setFromDlq(fromDlq);
+        caseEventMessageEntity.setState(state);
+        caseEventMessageEntity.setMessageProperties(getMessageProperties(message));
+        caseEventMessageEntity.setMessageContent(message);
+        caseEventMessageEntity.setReceived(LocalDateTime.now());
+        caseEventMessageEntity.setDeliveryCount(0);
+        caseEventMessageEntity.setRetryCount(0);
+
+        return caseEventMessageEntity;
+    }
+
+    private CaseEventMessageEntity build(String messageId,
+                                         String message,
+                                         boolean fromDlq,
+                                         MessageState state) {
+        CaseEventMessageEntity caseEventMessageEntity = new CaseEventMessageEntity();
+        caseEventMessageEntity.setMessageId(messageId);
+        caseEventMessageEntity.setFromDlq(fromDlq);
+        caseEventMessageEntity.setState(state);
+        caseEventMessageEntity.setMessageContent(message);
+        caseEventMessageEntity.setReceived(LocalDateTime.now());
+        caseEventMessageEntity.setDeliveryCount(0);
+        caseEventMessageEntity.setRetryCount(0);
+
+        return caseEventMessageEntity;
+    }
+
+    private JsonNode getMessageProperties(String message) throws JsonProcessingException {
+        JsonNode messageAsJson = objectMapper.readTree(message);
+        return messageAsJson.findPath(MESSAGE_PROPERTIES);
+    }
+
+    private boolean validate(String messageId, EventInformation eventInformation) {
+        // check all required fields
+        log.info("Validating message with id '{}'", messageId);
+        return isNotBlank(eventInformation.getCaseId()) && eventInformation.getEventTimeStamp() != null;
```

Comment:
Can we test to cover this line?

Reply:
We do have an integration test for this line. Gareth's PR will bring an unit test for the whole class.

Reply:
Here: https://github.com/hmcts/wa-case-event-handler/pull/199/files#diff-2f0e83622d53741233a2396b7d0462f88c2ba8d02ed39b11c2eeefed1cf22d9aR197

Reply:
Here in folded ControllerEndpointTest.java: https://github.com/hmcts/wa-case-event-handler/pull/199/files#diff-2f0e83622d53741233a2396b7d0462f88c2ba8d02ed39b11c2eeefed1cf22d9aR230

Reply:
Fixed by removing `eventInformation.getEventTimeStamp() != null`